libopenage/gui/guisys/private/gui_rendering_setup_routines.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+#include "gui_rendering_setup_routines.h"
+
+#include <cassert>
+
+#include <QDebug>
+
+#include "gui_ctx_setup.h"
+
+namespace qtsdl {
+
+GuiRenderingSetupRoutines::GuiRenderingSetupRoutines(SDL_Window *window) {
+	try {
+		this->ctx_extraction_mode = std::make_unique<GuiUniqueRenderingContext>(window);
+	} catch (const CtxExtractionException&) {
+
+		qInfo() << "Falling back to separate render context for GUI";
+
+		try {
+			this->ctx_extraction_mode = std::make_unique<GuiSeparateRenderingContext>(window);
+		} catch (const CtxExtractionException&) {
+			assert(false && "setting up context for GUI failed");
```

Comment:
The whole assert or the comment in it?

That's what I called "replace the exception with an assert".

It's kind of a spiritual successor to this one:

https://github.com/SFTtech/openage/blob/816fa62150da6e6c17461b538c4f8888f8b2519d/libopenage/gui/guisys/private/platforms/context_extraction_x11.cpp#L26